builtin/mainmenu/dlg_contentstore.lua

```diff
@@ -351,20 +351,24 @@ function store.get_formspec()
 	end
 
 	local formspec = {
-		"size[12,6.5;true]",
-		"field[0.3,0.1;10.2,1;search_string;;", core.formspec_escape(search_string), "]",
-		"field_close_on_enter[search_string;false]",
-		"button[10.2,-0.2;2,1;search;", fgettext("Search"), "]",
-		"dropdown[0,1;2.4;type;",
-		table.concat(filter_types_titles, ","),
-		";",
-		filter_type,
-		"]",
-		-- "textlist[0,1;2.4,5.6;a;",
-		-- table.concat(taglist, ","),
-		-- "]"
+		"size[12,6.5;true]"
 	}
 
+	if #store.packages ~= 0 then
+		formspec[#formspec + 1] = "field[0.3,0.1;10.2,1;search_string;;"
```

Comment:
it would be better to initialise the list here. ie:

```lua
local formspec
if #store.packages ~= 0 then
    formspec = {
        ---
    }
else
    formspec = {
        ---
    }
end
```

Reply:
Since `size[]` is required irrespective of whether packages can be retrieved or not, which is why I initialized `formspec` with a `size[]` element. Should I add the `size[]` element to both the initializers instead?

It'd actually look more cleaner than adding `formspec[#formspec + 1]` at the start of every line...

Reply:
Yes. It's also more efficient

Reply:
Seems unresolved and not done.

Reply:
I'm sorry, my bad. Will fix ASAP.

Reply:
Done. @paramat